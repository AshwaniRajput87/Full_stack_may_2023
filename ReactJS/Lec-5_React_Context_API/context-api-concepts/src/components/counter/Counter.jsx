import { useCounter } from '../counterProvider/CounterProvider'

const Counter = () => {

    /***
     * Here we are consuming the context
     */
    const { increment } = useCounter();

    return (
        <>
           <button onClick={increment}>Increment</button>
        </>
    )
}

export default Counter;