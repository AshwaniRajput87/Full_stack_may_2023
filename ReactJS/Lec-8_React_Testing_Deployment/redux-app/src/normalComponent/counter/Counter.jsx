import React, { useState } from 'react';

const Counter = () => {
    const [count, setCount] = useState(0);
    
    const handleDecrement = () => {
        setCount(count - 1);
    }

    const handleIncrement = () => {
        setCount(count + 1);
    }
     
    return (
        <>
          <h2>Counter</h2>
           <div className='container'>
                <button data-test-id="counter-decrement-btn" onClick={handleDecrement}>-</button>
                <p>{count}</p>
                <button onClick={handleIncrement}>+</button>
           </div>
           <div>
               Text
           </div>
        </>
    )
}

export default Counter;