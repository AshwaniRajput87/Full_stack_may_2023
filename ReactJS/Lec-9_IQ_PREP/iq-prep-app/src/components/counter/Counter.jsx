import { PureComponent } from 'react';

class Counter extends PureComponent {

    constructor(props) {
        super(props);
        this.state = {
            count: 0,
            // x: 10
        }
        this.handleIncrement = this.handleIncrement.bind(this);
    }

    componentDidMount(){
       console.log('Component gets mounted');
    }

    componentDidUpdate(prevProps, prevState){
        console.log('Component gets updated, Count', prevState.count);
    }

    componentWillUnmount(){
        console.log('Component gets unmounted');
    }

    handleIncrement() {
        this.setState((prevState)=>({
            count: prevState.count + 1
        }));

        // console.log('handle click');
    }

    render(){
        return (
            <>
             <p>Count: {this.state.count}</p>
             <button onClick={this.handleIncrement}>Increment</button>
            </>
        )
    }

}

export default Counter;