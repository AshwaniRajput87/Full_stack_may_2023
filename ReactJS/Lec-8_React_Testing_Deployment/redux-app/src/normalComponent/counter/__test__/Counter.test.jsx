
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Counter from '../Counter';

describe('Counter component', ()=> {
    test('snapshot testing', ()=>{
       const { asFragment } =  render(<Counter />);
       expect(asFragment()).toMatchSnapshot();
    });

    test('renders intial count of 0', ()=> {
        render(<Counter />);
       const decrementBtn = screen.getByText('-');
       const countText = screen.getByText('0');
       const incrementBtn = screen.getByText('+');

       expect(decrementBtn).toBeTruthy();
       expect(incrementBtn).toBeTruthy();
       expect(countText).toBeTruthy();

    });

    test('increment button increases the count by 1', ()=> {
        render(<Counter />);
        const incrementBtn = screen.getByText('+');
        fireEvent.click(incrementBtn);

        const countTxt = screen.getByText('1');
        expect(countTxt).toBeTruthy();
    });

    // Please write the decrement button test here. -> HW
});