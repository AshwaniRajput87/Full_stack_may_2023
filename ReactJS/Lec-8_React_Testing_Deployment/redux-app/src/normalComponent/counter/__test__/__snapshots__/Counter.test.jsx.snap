// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component snapshot testing 1`] = `
<DocumentFragment>
  <h2>
    Counter
  </h2>
  <div
    class="container"
  >
    <button
      data-test-id="counter-decrement-btn"
    >
      -
    </button>
    <p>
      0
    </p>
    <button>
      +
    </button>
  </div>
  <div>
    Text
  </div>
</DocumentFragment>
`;
